### PR TITLE
Add cni plugins

### PR DESCRIPTION
**Pull request by varder**
CNI - the Container Network Interface consists of a specification and
libraries for writing plugins to configure network interfaces in Linux
containers
aos-firewall plugin handles access to containers own by different service providers
dnsname plugin - provides DNS services and service discovery for local containers  
@al1img @oleksiimoisieiev

### CODE REVIEW

File: recipes-networking/cni/dnsname/0001-resolve-local-hostnames.patch [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+-	"path/filepath"
+-
+ 	"github.com/containernetworking/cni/pkg/skel"
+ 	"github.com/containernetworking/cni/pkg/types"
+ 	"github.com/containernetworking/cni/pkg/types/current"
+@@ -40,6 +36,10 @@ import (
+ 	bv "github.com/containernetworking/plugins/pkg/utils/buildversion"
+ 	"github.com/pkg/errors"
+ 	"github.com/sirupsen/logrus"
++	"io/ioutil"
```
Comment (oleksiimoisieiev) [inline recipes-networking/cni/dnsname/0001-resolve-local-hostnames.patch:203]: This change is unnecessary.

File: recipes-networking/cni/dnsname/0001-resolve-local-hostnames.patch [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+ package main
+ 
+ import (
++	"bytes"
+ 	"encoding/json"
+-	"io/ioutil"
+-	"os"
+-	"os/exec"
+-	"path/filepath"
+-
```
Comment (oleksiimoisieiev) [inline recipes-networking/cni/dnsname/0001-resolve-local-hostnames.patch:195]: This change is unnecessary.

**Review by oleksiimoisieiev — CHANGES_REQUESTED** [review]

File: recipes-networking/cni/aos-firewall.bb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    ${GO} build -o ${B}/bin/aos-firewall
+}
+
+do_install() {
+    localbindir="${libexecdir}/cni/"
+    install -d ${D}/${localbindir}
+    install -m 755 ${B}/bin/aos-firewall ${D}/${localbindir}
+}
+
+FILES_${PN} = "${libexecdir}/cni/"
```
Comment (oleksiimoisieiev) [inline recipes-networking/cni/aos-firewall.bb:24]: use common variable for ${libexecdir}/cni/

File: recipes-networking/cni/aos-firewall.bb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+SRCREV = "${AUTOREV}"
+
+LICENSE = "Apache-2.0"
+LIC_FILES_CHKSUM = "file://src/${GO_IMPORT}/LICENSE;md5=3b83ef96387f14655fc854ddc3c6bd57"
+
+SRC_URI = "git://gitpct.epam.com/Valerii_Chubar/aos-filter-firewall.git;protocol=https;branch=core-logic;" 
+
+do_compile() {
+    cd ${S}/src/aos-firewall/plugins/meta/aos-firewall
+    ${GO} build -o ${B}/bin/aos-firewall
```
[the remark this answers is not in the document]
Comment (varder) [reply recipes-networking/cni/aos-firewall.bb:15]: It seems that makefile is redundant here. l suggest to update aos-firewall directory structure instead

**Comment by al1img** [comment]
Added in meta-aos

**Pull request closed without merging**